FtpClient/FtpCli/InitializeSession.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+
+namespace FtpCli
+{
+  // The IntializeSession class prompts the user for session details
+  // which are stored as a Data object
+  public class InitializeSession
+  {
+      Data data;
+
+      public InitializeSession() {
+        data = new Data();
+      }
+
+    // prompt user for server, username, and commands (if provided)
+    public Data initialPrompt(string[] args) {
+      string response = "";
+
+      do {
+        response = this.promptUser("Enter Server: ");
+      } while (!validateNonEmptyResponse(response));
+      this.data.setServer(response);
+
+      do {
+        response = this.promptUser("Enter Username: ");
+      } while (!validateNonEmptyResponse(response));
```

Comment:
What if they have leading/trailing spaces in their response? Do we want to trim that or roundly deny if the input isn't exact?

Reply:
Nice spot -- adding a fix for that.